### PR TITLE
get parent oci component information when corgi download_url failed and no parent info found

### PR DESCRIPTION
**Pull request by RichardXuan**
Fixes OLCS-594

### CODE REVIEW

File: openlcsd/flow/tasks.py [hunk 1]
```diff
@@ -293,23 +293,62 @@ def download_shared_remote_source(context, engine):
...
+        for source in sources:
+            if not source['purl'].startswith("pkg:oci"):
+                continue
+            # get build complete time from brew, get the newest one
+            oci_component = connector.get(
+                source['link'],
+                includes=connector.default_includes_type_independence
+            )
+            build_info = koji_connector.get_build(
+                int(oci_component["software_build"]['build_id']))
```
Comment (rhyw) [inline openlcsd/flow/tasks.py:320]: this assumes `oci_component` returned from corgi always have non-empty software_build.

File: openlcsd/flow/tasks.py [hunk 2]
```diff
@@ -293,23 +293,62 @@ def download_shared_remote_source(context, engine):
...
+            completion_ts = build_info['completion_ts']
```
Comment (rhyw) [inline openlcsd/flow/tasks.py:321]: completion_ts could be null according to brew's api docs.

File: openlcsd/flow/tasks.py [hunk 3]
```diff
@@ -293,23 +293,62 @@ def download_shared_remote_source(context, engine):
...
+            if completion_ts > oci_ts:
```
Comment (rhyw) [inline openlcsd/flow/tasks.py:322]: not sure I could get it for this comparion. would it possible to have `completion_ts ` < 0?

File: openlcsd/flow/tasks.py (hunk 1 above)
Comment (RichardXuan) [reply openlcsd/flow/tasks.py:320]: Yes. Is there any chance oci component have empty software_build field.

File: openlcs/libs/corgi.py [hunk 4]
```diff
@@ -159,6 +159,17 @@ def rpmmod_includes(self):
     def default_includes(self):
         return self.get_include_fields()
 
+    @functools.cached_property
+    def default_includes_type_independence(self):
+        """
+        get default includes field list, type independence.
```
[the remark this answers is not in the document]
Comment (RichardXuan) [reply openlcs/libs/corgi.py:165]: My intention is to retrieve a minimal list of fields for a component, type independence. Cause i notice these are some field i do not need in this situation when use get_include_fields(component_type="oci").  Want reduce memory consume.

File: openlcsd/flow/tasks.py (hunk 3 above)
Comment (RichardXuan) [reply openlcsd/flow/tasks.py:322]: creation_ts: time the build was created (epoch)

I think i can change to use creation_ts field, this field will not be null. And i think i will not < 0.

File: openlcs/libs/corgi.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (rhyw) [reply openlcs/libs/corgi.py:165]: it's not clear to me what `type independence` means. type IMO should be consisent with corgi enums.

File: openlcsd/flow/tasks.py (hunk 1 above)
Comment (rhyw) [reply openlcsd/flow/tasks.py:320]: I didn't find one right now. but I think it is literally possible to have one with `namespace` other than `REDHAT`.

I think it's ok to keep it this way for now.

File: openlcs/libs/corgi.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (RichardXuan) [reply openlcs/libs/corgi.py]: Updated, please have a look

**Comment by rhyw** [comment]
Thanks for the update. generally LGTM.

**Pull request merged**